graphql-spring-boot-test/src/main/java/com/graphql/spring/boot/test/RequestFactory.java

```diff
@@ -19,8 +19,8 @@ private RequestFactory() {
     static HttpEntity<Object> forMultipart(String query, String variables, HttpHeaders headers) {
         headers.setContentType(MediaType.MULTIPART_FORM_DATA);
```

Comment:
It's setting the Content-Type header explicitly to `MULTIPART_FORM_DATA` here, so it should be good actually.

Reply:
Nope because you pass the same map of headers to the `forJson` method which overrides the `Content-Type` header to `application/json`.

Reply:
I'll have to see how these methods are being used exactly then to understand what you mean. Because in that case it sounds like the parameter `header` should be removed from this method entirely.

Reply:
Can you share your test code where you use this to demonstrate the problem you're facing here? Because `GraphQLTestTemplate.postMultipart` only updates the content-type header here just before it uses `restTemplate.exchange`, so that should be good imo. `forJson` does exactly the same. So when you call `forJson` first followed by `forMultipart` or the other way around from your tests I'd expect it to work properly. The problem you have that led you to this solution is not clear to me now.

Reply:
The test code does nothing fancy, a simple call to `graphQLTestTemplate.postMultipart()` triggers the error. I don't use the factory methods from `RequestFactory` since they are package private.

The problem is `graphQLTestTemplate.postMultipart()` calls the factory method `RequestFactory.forMultipart` which in turns calls the factory method `RequestFactory.forJson` with the same map of headers, which overrides the `Content-Type` set in `RequestFactory.forMultipart`.

Reply:
Ah, I totally missed it was calling the `forJson` from within the `forMultipart` method.